applications/GeoMechanicsApplication/tests/cpp_tests/test_backward_euler_UPw_scheme.cpp

```diff
@@ -0,0 +1,124 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Richard Faasse
+//
+
+#include "containers/model.h"
+#include "custom_strategies/schemes/backward_euler_quasistatic_U_Pw_scheme.hpp"
+#include "spaces/ublas_space.h"
+#include "testing/testing.h"
+
+namespace Kratos::Testing
+{
+
+using namespace Kratos;
+using SparseSpaceType = UblasSpace<double, CompressedMatrix, Vector>;
+using LocalSpaceType = UblasSpace<double, Matrix, Vector>;
+
+class BackwardEulerUPwSchemeTester
+{
+public:
+    Model mModel;
+    BackwardEulerQuasistaticUPwScheme<SparseSpaceType, LocalSpaceType> mScheme =
+        CreateValidScheme();
+
+    BackwardEulerUPwSchemeTester() { CreateValidModelPart(); }
+
+    BackwardEulerQuasistaticUPwScheme<SparseSpaceType, LocalSpaceType> CreateValidScheme() const
+    {
+        return BackwardEulerQuasistaticUPwScheme<SparseSpaceType, LocalSpaceType>();
+    }
+
+    void CreateValidModelPart()
+    {
+        auto& result = mModel.CreateModelPart("dummy", 2);
+        result.AddNodalSolutionStepVariable(VELOCITY);
+        result.AddNodalSolutionStepVariable(ACCELERATION);
+        result.AddNodalSolutionStepVariable(DT_WATER_PRESSURE);
+        result.AddNodalSolutionStepVariable(DISPLACEMENT);
+        result.AddNodalSolutionStepVariable(WATER_PRESSURE);
+
+        auto p_node = result.CreateNewNode(0, 0.0, 0.0, 0.0);
+        p_node->AddDof(DISPLACEMENT_X);
+        p_node->AddDof(DISPLACEMENT_Y);
+        p_node->AddDof(DISPLACEMENT_Z);
+        p_node->AddDof(WATER_PRESSURE);
+        result.GetProcessInfo()[DELTA_TIME] = 4.0;
+
+        p_node->FastGetSolutionStepValue(VELOCITY, 1) =
+            Kratos::array_1d<double, 3>{1.0, 2.0, 3.0};
+        p_node->FastGetSolutionStepValue(ACCELERATION, 1) =
+            Kratos::array_1d<double, 3>{4.0, 5.0, 6.0};
+        p_node->FastGetSolutionStepValue(DISPLACEMENT, 1) =
+            Kratos::array_1d<double, 3>{7.0, 8.0, 9.0};
+
+        p_node->FastGetSolutionStepValue(WATER_PRESSURE, 1) = 1.0;
+        p_node->FastGetSolutionStepValue(WATER_PRESSURE, 0) = 2.0;
+    }
+
+    ModelPart& GetModelPart() { return mModel.GetModelPart("dummy"); }
+};
+
+KRATOS_TEST_CASE_IN_SUITE(CheckBackwardEulerUPwScheme_ReturnsZeroForValidScheme,
```

Comment:
Perhaps the following suggested name is slightly clearer?
```suggestion
KRATOS_TEST_CASE_IN_SUITE(CheckBackwardEulerUPwScheme_ReturnsZeroForDefaultConstructedScheme,
```

Reply:
As discussed, I mention now in the name that it's about a valid modelpart, not about a valid scheme